src/note.cpp

```diff
@@ -8,24 +8,25 @@
 /*************************************************************************/
 
 #include "note.h"
+#include "CommandSet.hpp"
 #include "Descriptor.hpp"
 #include "buffer.h"
 #include "merc.h"
 #include "string_utils.hpp"
-#include "trie.h"
 
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
+#include <functional>
+#include <memory>
 
 static NOTE_DATA *note_first;
 static NOTE_DATA *note_last;
 
-static void *trie;
-
-typedef void(note_fn_t)(CHAR_DATA *, const char *);
+typedef std::function<void(CHAR_DATA *ch, const char *arg)> note_fn;
```

Comment:
note_fn could potentially reuse the one that's currently in interp.cpp line 45 as it's the same signature.

Reply:
Similar signature, yes (const here, non-const in interp.cpp). To reuse it, though, I'd have to move it to a .h file. Plus, I think that while they do have very similar signatures now, they don't necessarily refer to the same thing.